app_check/src/include/firebase/app_check/debug_provider.h

```diff
@@ -20,6 +20,12 @@
 namespace firebase {
 namespace app_check {
 
+/// @cond FIREBASE_APP_INTERNAL
```

Comment:
How does /// @cond work compared to #IFDEF ?

Would it be simpler to have a forward declaration at the top of this file?
Forward declarations is what I have seen for our public headers for classes like
StorageInternal, AppInternal, AuthData, InstallationsInternal, etc

Reply:
I believe it is more for doxygen, and for whatever reason we have been using it for other headers.  But we talked a bit offline, and it doesn't seem to be necessary, so will remove.